apps/src/JSInterpreter.js

```diff
@@ -856,30 +914,32 @@ JSInterpreter.prototype.getProperty = function (
  *
  * Does the named property exist on a data object.
  * @param {!Object} interpeter Interpreter instance.
- * @param {!Function} baseHasProperty Original hasProperty() implementation.
  * @param {!Object} obj Data object.
  * @param {*} name Name of property.
  * @return {boolean} True if property exists.
  */
 JSInterpreter.prototype.hasProperty = function (
     interpreter,
-    baseHasProperty,
     obj,
     name) {
   name = name.toString();
   var nativeParent;
-  if (obj.isCustomMarshal ||
-      (obj === this.globalScope &&
-          (!!(nativeParent = this.customMarshalGlobalProperties[name])))) {
-    if (this.shouldBlockCustomMarshalling_(name, obj, nativeParent)) {
+  if (obj.isCustomMarshal) {
+    if (this.shouldBlockCustomMarshalling_(name, obj)) {
       return false;
-    } else if (obj.isCustomMarshal) {
-      return name in obj.data;
     } else {
-      return name in nativeParent;
+      return name in obj.data;
     }
   } else {
-    return baseHasProperty.call(interpreter, obj, name);
+    var hasProperty = JSInterpreter.baseHasProperty.call(interpreter, obj, name);
+    if (!hasProperty &&
+        obj === this.globalScope &&
+        !!(nativeParent = this.customMarshalGlobalProperties[name]) &&
```

Comment:
The assignment works in the linter if you prefix it with the `!!` to make it clear that you want the Boolean for the comparison. This part is existing code and there was an attempt to factor it in a function before, but there were subtle differences. Now, it looks like it will be easier to do so. So I'll resolve both in my next PR.